src/components/MainTable/index.js

```diff
@@ -0,0 +1,188 @@
+        sessionStorage.setItem('display', JSON.stringify(displayedFilteredArr))
+
+        dispatch(deleteFromPinned(pinnedFilteredArr))
+        sessionStorage.setItem('pinned', JSON.stringify(pinnedFilteredArr))
+    }
+
+    const dragStartHandler = (e, country) => {
+        setCurrentCard(country)
+    }
+
+    const dragOverHandler = e => {
+        e.preventDefault()
+    }
+
+    const dropHandler = (e, country) => {
+        e.preventDefault()
+        dispatch(showDisplayedCountries(displayedCountries.map( item => {
+            if (item.id === country.id) {
+                return {...item, order: currentCard.order}
+            }
+            if (item.id === currentCard.id) {
+                return {...item, order: country.order}
+            }
+            return item
+        })))
+    }
+
+    const sortCards = (a, b) => {
+        if (a.order > b.order) {
+            return 1
+        } else {
+            return -1
+        }
+    }
+
+
+    return (
+        <div className='main-wrapper'>
+            <div className='input-wrapper'>
+                <CustomInput 
+                    placeholder='Enter country name..'
+                    value={input}
+                    onChange={e => onChangeHandler(e)}
+                    />
+                <CustomButton 
+                    name='X'
+                    onClick={onClickHandler}
+                    style={{fontSize: '1.5rem'}}
+                />
+            </div>
+            <div  className='countries-wrapper'>
+                <Flex justify='center' direction='row'>
+                    {pinnedCountries ? pinnedCountries.map((country, index) => (
+                        <CountryCard
+                            path={'/details?id='+country.id}
+                            deleteHandler={deleteHandler}
+                            key={index}
+                            onChange={pinHandler}
+                            country={country}
+                        />)
+                    ) : null}
+                    {displayedCountries ? displayedCountries.sort(sortCards).map((country, index) => (
+                        <div 
+                            onDragStart={e => dragStartHandler(e, country)}
+                            onDragOver={e => dragOverHandler(e)}
+                            onDrop={e => dropHandler(e, country)}
+                            draggable={true}
+                            key={index}
+                        >
+                            <CountryCard
+                                path={'/details?id='+country.id}
+                                deleteHandler={deleteHandler}
+                                onChange={pinHandler}
+                                country={country}
+                            />
+                        </div>
+                    )) : null}
+                </Flex>
+            </div>
+        </div>
+    )
+}
+
+export default MainTable
```

Comment:
It is better to split  big components to smaller components or move handling logic to custom hooks